controllers/users.go

```diff
@@ -1,61 +1,94 @@
 package controller
 
 import (
-    "net/http"
+	"net/http"
 
-    "github.com/gin-gonic/gin"
-    "github.com/uptrace/bun"
-    "github.com/Real-Dev-Squad/tiny-site-backend/models"
+	"github.com/Real-Dev-Squad/tiny-site-backend/dtos"
+	"github.com/Real-Dev-Squad/tiny-site-backend/models"
+	"github.com/gin-gonic/gin"
+	"github.com/uptrace/bun"
 )
 
 func GetUserList(ctx *gin.Context, db *bun.DB) {
-    var users []models.User
-    err := db.NewSelect().Model(&users).OrderExpr("id ASC").Limit(10).Scan(ctx)
-
-    if err != nil {
-        ctx.JSON(http.StatusInternalServerError, gin.H{
-            "message": "error",
-        })
-        return
-    }
-
-    ctx.JSON(http.StatusOK, gin.H{
-        "message": "users fetched successfully",
-        "data":    users,
-    })
+	var users []models.User
+	err := db.NewSelect().Model(&users).OrderExpr("id ASC").Limit(10).Scan(ctx)
+
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, dtos.UserListResponse{
+			Message: "Failed to fetch users: " + err.Error(),
+		})
+		return
+	}
+
+	if len(users) == 0 {
+		ctx.JSON(http.StatusNotFound, dtos.UserListResponse{
+			Message: "No users found",
+		})
+		return
+	}
+
+	var dtoUsers []dtos.User
+	for _, user := range users {
+		dtoUsers = append(dtoUsers, dtos.User{
+			ID:        user.ID,
+			Email:     user.Email,
+			CreatedAt: user.CreatedAt,
+		})
+	}
+
+	ctx.JSON(http.StatusOK, dtos.UserListResponse{
+		Message: "users fetched successfully",
+		Data:    dtoUsers,
+	})
```

Comment:
For single data we should return only user data not like this

Reply:
I fixed this type of issue in previous PR

Reply:
What type i have mentioned in description a screenshot so do checkout

Reply:
> I fixed this type of issue in previous PR

That issue is different how that issue is link to this that related to error message

Reply:
For single data i want a response like this ![285787289-534b37cf-15e8-4e95-bfe8-636aca8024da.png](https://github.com/Real-Dev-Squad/tiny-site-backend/assets/70854507/969f2832-6a7d-4f4c-8bd9-4e009bc4a3e4)

No message if data is returned successfully

Reply:
![image](https://github.com/Real-Dev-Squad/tiny-site-backend/assets/111434418/0b93ff80-bebb-46eb-ae78-09fe8e028063)

Why no message we can returned message as it provides more context
as we can remove but I don't thing is require to remove the message

Reply:
Please ask  Ankush in the morning standup if possible

Reply:
I talk about data object should be there as json spec is saying that i have mentioned the  screenshot above and message can be optional

Reply:
Let's wait for Prakash and Randhir what's their point over this

Reply:
Some of the APIs on our backend return data with the `data` key and the corresponding response, while others directly send the data in the response body as @sahsisunny mentioned users api self response. This variation depends on how the consumer or frontend client is consuming the API. Neither of these approaches follows the JSON API specification so it will be recommended to follow the format that our frontend expects the data to be in.